qiskit/circuit/library/evolved_operator_ansatz.py

```diff
@@ -145,20 +133,7 @@ def operators(self, operators=None) -> None:
         operators = _validate_operators(operators)
         self._invalidate()
         self._operators = operators
-
-    @property
-    def qregs(self):
-        """A list of the quantum registers associated with the circuit."""
-        if self._data is None:
-            self._build()
-        return self._qregs
-
-    @qregs.setter
-    def qregs(self, qregs):
-        """Set the quantum registers associated with the circuit."""
-        self._qregs = qregs
-        self._qubits = [qbit for qreg in qregs for qbit in qreg]
-        self._invalidate()
+        self.qregs = [QuantumRegister(self.num_qubits, name="q")]
```

Comment:
If we use `add_register` then `qregs` and `qubits` are out of sync (I implemented it again to try 😄) so until we fix that I think we'll have to use the above 🙂

Reply:
That made (makes!) me very nervous, but it turns out the reason is because `BlueprintCircuit.qregs` is a property whose setter internally calls `add_register`.  So the use in this PR is ok (even though `BlueprintCircuit` in general makes me very nervous!).